products/llbuildSwift/CMakeLists.txt

```
@@ -16,7 +16,7 @@ if(CMAKE_VERSION VERSION_LESS 3.16)
 endif()
 
 # Set x86_64 default if CMAKE_OSX_ARCHITECTURES is not set
-if(NOT DEFINED CMAKE_OSX_ARCHITECTURES)
+if(NOT DEFINED CMAKE_OSX_ARCHITECTURES OR CMAKE_OSX_ARCHITECTURES STREQUAL "")
   set(CMAKE_OSX_ARCHITECTURES "x86_64")
```

Could this maybe use current architecture instead of hardcoding it? This would make it compatible with Apple Silicon as I'm suggesting here https://github.com/apple/swift/pull/34998/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a

```suggestion
   set(CMAKE_OSX_ARCHITECTURES "${CMAKE_HOST_SYSTEM_PROCESSOR}")
```

---

Seems reasonable!

---

I will create a new PR for this change to avoid having to re-run testing on this PR.

---

https://github.com/apple/swift-llbuild/pull/709